### PR TITLE
test(bingx): scope trade side-sequence exception to swaps

### PR DESCRIPTION
**Pull request by AresArtemius**
## Summary

BingX swap recent trades can contain same-millisecond fills in non-monotonic price order for the same taker side. In a captured public response, buy fills 727492266 and 727492255 at timestamp 1789587177591 arrive at prices 75598.6 and 75590.0. CCXT preserves their input order. The funding changes in #30471 do not touch this path.

- Add an opt-in `sideSequenceSwap` test setting, honoured only for resolved swap markets, and enable it for BingX.
- Keep spot sequence validation and trade structure, timestamp, side and both-side checks enabled.
- Do not change exchange parsing or sort by undocumented fillId sequencing.
- Add seven native JS regression tests and a dedicated JS CI step, including a flag-removal negative control.

## Verification

- TypeScript build and scoped ESLint passed.
- JS regressions: 7 passed; BingX static request: 190 passed; static response: 53 passed.
- Python sync/async, PHP sync and Java: offline swap acceptance, spot rejection and flag-removal negative control passed in each lane.
- PHP sync/async generated syntax checks passed; PHP async runtime was not exercised.
- C#, Java and Go test builds passed. C#/Go behavioral replay was not performed.
- Only handwritten source/test/workflow/config files included; no generated output.

## Rust prerequisite: #30470

Without the common Rust fix, local `cargo check` fails on missing `Value.group_by` and an awaited helper emitted as a synchronous function. Temporarily applying only the Rust generator/helper/test changes from #30470 makes all three generator regression tests and `cargo check` pass with this candidate. The temporary source changes were then restored and are not included here.

This is compilation verification, not a Rust behavioral replay. The Rust prerequisite should land before expecting this branch's Rust CI to pass. No claim of hosted CI success.

### CODE REVIEW

**Review by carlotestor — COMMENTED** [review]
## Summary

Scopes the `fetchTrades` side-sequence assertion so it can be skipped for resolved swap markets only, via a new `sideSequenceSwap` key, and enables it for BingX. Spot keeps the original assertion, and the other structural checks (side, timestamp order, both-sides) stay on. The regression tests include a flag-removal negative control and a cross-exchange control, which is the right shape for a test-gating change — it proves the exception is narrow rather than just proving the suite goes green.

### Findings

- `exchange.market (symbol)` is now called unconditionally at the top of `testFetchTrades`, for every exchange, whether or not the new key is present. This file is transpiled to Python/PHP/C#/Go/Java, so any symbol that does not resolve cleanly now raises from `market()` before the existing assertions run, instead of the behaviour being unchanged for exchanges that never opt in. Resolving the market only when `'sideSequenceSwap' in skippedProperties` keeps the no-opt-in path byte-for-byte identical.
- The new CI step runs unconditionally, while the neighbouring suites are gated on `env.important_modified`. It works because both `pre-transpile-js` and `pre-transpile-js-simple` run `tsc` first, so `js/src/...` exists in either branch — worth stating explicitly, since the `.mjs` imports the compiled tree and would fail confusingly if that step order ever changes.

Nothing here looks merge-blocking: the production parsing path is untouched and the skip is opt-in per exchange plus per market type.

Merge gate: 🟢

File: ts/src/test/Exchange/test.fetchTrades.ts [hunk 1]
```diff
@@ -34,7 +34,9 @@ async function testFetchTrades (exchange: Exchange, skippedProperties: object, s
     if (!('timestampSort' in skippedProperties)) {
         testSharedMethods.assertTimestampOrder (exchange, method, symbol, trades);
     }
-    if (!('side' in skippedProperties) && !('sideSequence' in skippedProperties)) {
+    const market = exchange.market (symbol);
+    const skipSwapSideSequence = (market['swap'] === true) && ('sideSequenceSwap' in skippedProperties);
```
Comment (carlotestor) [inline ts/src/test/Exchange/test.fetchTrades.ts]: `exchange.market (symbol)` now runs for every exchange on every `fetchTrades` test, even when no exchange opted into `sideSequenceSwap`. This file transpiles to Py/PHP/C#/Go/Java, so a symbol that does not resolve would now throw here before the existing side/timestamp assertions. Consider computing it lazily: `const skipSwapSideSequence = ('sideSequenceSwap' in skippedProperties) && (exchange.market (symbol)['swap'] === true);` — or an `if` guard — so the non-opt-in path is unchanged.

File: .github/workflows/js.yml [hunk 2]
```diff
@@ -114,6 +114,8 @@ jobs:
             npm run lint "ts/src/pro/$exchange.ts"
           fi
         done
+    - name: Test swap-only trade sequence skips
+      run: node --test build/tests/test.fetch-trades-side-sequence.mjs
```
Comment (carlotestor) [inline .github/workflows/js.yml:120]: This step is unconditional while the surrounding suites are gated on `env.important_modified`, and the `.mjs` imports `js/src/bingx.js` + `js/src/test/Exchange/test.fetchTrades.js`. That holds today because both `pre-transpile-js` and `pre-transpile-js-simple` run `tsc`, but it makes the step order load-bearing — worth placing it after the existing build-dependent steps (or gating it the same way) so it cannot silently become a module-not-found failure.

File: ts/src/test/Exchange/test.fetchTrades.ts (hunk 1 above)
Comment (AresArtemius) [reply ts/src/test/Exchange/test.fetchTrades.ts]: Fixed: market resolution now runs only when sideSequenceSwap is present. Added a regression proving market() is not called without opt-in.

File: .github/workflows/js.yml (hunk 2 above)
Comment (AresArtemius) [reply .github/workflows/js.yml:120]: Added a comment documenting the build dependency. The step remains after both pre-transpile branches, each of which runs tsc, and intentionally covers both full and scoped builds.

File: build/tests/test.fetch-trades-side-sequence.mjs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+test('without the opt-in flag the test does not resolve the market', async () => {
+    const { exchange, trades, run } = scenario('BTC/USDT:USDT', 'kraken');
+    trades.reverse(); // Valid increasing-price buy sequence.
+    let marketCalls = 0;
+    exchange.market = () => {
+        marketCalls++;
+        throw new Error('Unexpected market resolution without sideSequenceSwap');
+    };
+    await run();
+    assert.equal(marketCalls, 0);
```
Comment (carlotestor) [inline build/tests/test.fetch-trades-side-sequence.mjs:50]: `assert.equal(marketCalls, 0)` can never actually fail: the stub throws on the first call, so a regression surfaces as `run()` rejecting with `Unexpected market resolution without sideSequenceSwap` and this line is unreachable. That still catches the regression, but if you want the counter assertion to be the thing that reports, have the stub record and return a plausible market (`{ 'swap': true }`) instead of throwing.

**Review by carlotestor — COMMENTED** [review]
Follow-up on 9a5ca07d8e1755081748664674f62f24b9aa3217.

The delta does exactly what the previous pass asked on the transpiled test: `exchange.market (symbol)` is now resolved only inside `if ('sideSequenceSwap' in skippedProperties)`, so the non-opt-in path through `testFetchTrades` is unchanged for every other exchange in Py/PHP/C#/Go/Java. The added negative control (`without the opt-in flag the test does not resolve the market`) pins that behaviour rather than just asserting the suite is green, which is the right way to keep the guard from regressing.

The CI step keeps its unconditional placement and now carries a comment naming the invariant it relies on (both `pre-transpile-js` and `pre-transpile-js-simple` run `tsc`, and the step sits after both branches). That is a reasonable resolution of the earlier note — the load-bearing ordering is documented instead of implicit.

One small test-mechanics note inline; not blocking.

Merge gate: 🟢

File: build/tests/test.fetch-trades-side-sequence.mjs (hunk 3 above)
Comment (AresArtemius) [reply build/tests/test.fetch-trades-side-sequence.mjs:50]: The throwing stub intentionally fails immediately if market() is called without opt-in. Agreed that the counter assertion is redundant; the regression is still covered.

**Comment by carlosmiei** [comment]
@AresArtemius we shouldn't add ad hoc tests not aligned with our unified testing framework, otherwise we break the test parity across langs
